app/components/custom_list/user_list_row/user_list_row.js

```diff
@@ -17,36 +18,62 @@ import {displayUsername} from 'mattermost-redux/utils/user_utils';
 export default class UserListRow extends React.PureComponent {
     static propTypes = {
         id: PropTypes.string.isRequired,
+        currentUserId: PropTypes.string.isRequired,
         theme: PropTypes.object.isRequired,
         user: PropTypes.object.isRequired,
         teammateNameDisplay: PropTypes.string.isRequired,
         ...CustomListRow.propTypes
     };
 
+    static contextTypes = {
+        intl: intlShape
+    };
+
     onPress = () => {
         this.props.onPress(this.props.id);
     };
 
     render() {
-        const style = getStyleFromTheme(this.props.theme);
+        const {formatMessage} = this.context.intl;
+        const {
+            currentUserId,
+            enabled,
+            onPress,
+            selectable,
+            selected,
+            teammateNameDisplay,
+            theme,
+            user
+        } = this.props;
+
+        const {id, username} = user;
+        const style = getStyleFromTheme(theme);
+
+        let usernameDisplay = `(@${username})`;
+        if (id === currentUserId) {
+            usernameDisplay = formatMessage({
+                id: 'mobile.more_dms.you',
+                defaultMessage: '(@{username} - you)'
+            }, {username});
+        }
 
         return (
             <CustomListRow
-                id={this.props.id}
-                theme={this.props.theme}
-                onPress={this.props.onPress ? this.onPress : null}
-                enabled={this.props.enabled}
-                selectable={this.props.selectable}
-                selected={this.props.selected}
+                id={id}
+                theme={theme}
+                onPress={onPress ? this.onPress : null}
```

Comment:
I think you should pass `this.onPress` and in the function validate that the prop exists